make/env.mk

```
@@ -40,10 +40,15 @@ endif
 TAG := # make sure tag is never injectable as an env var
 RELEASE_GOTAGS := release
 
-# Use a release go -tag when CI is targetting a tag
+# Use a release go -tag when CI is targeting a tag
 ifdef CI
 ifneq ($(BUILD_TAG),)
+ifeq ($(GOTAGS),)
 GOTAGS := $(RELEASE_GOTAGS)
+else
+# Preserve existing GOTAGS and append release tags
+GOTAGS := $(GOTAGS),$(RELEASE_GOTAGS)
+endif
```

Let's please use here one-liner that you arrived in another commit.
Also, if you check around, declaring `comma` is an established practice (n=2) so you may not need to assist that with so many explanations.
https://github.com/stackrox/stackrox/blob/38e601a255f537320e69b3a13af01e91e4cc1741/Makefile#L130-L133

Untested:
```suggestion
GOTAGS := $(if $(GOTAGS),$(GOTAGS)$(comma))$(RELEASE_GOTAGS)
```